AmplifyPlugins/Core/AWSPluginsCore/Auth/Provider/AmplifyAWSCredentialsProvider.swift

```diff
@@ -27,10 +27,14 @@ extension AWSCredentials {
 
     func toAWSSDKCredentials() -> AWSClientRuntime.AWSCredentials {
         if let tempCredentials = self as? AWSTemporaryCredentials {
+
+            let expirationTimeSinceNow = tempCredentials.expiration.timeIntervalSinceNow
+            let expirationTimeout = UInt64(expirationTimeSinceNow < 0 ? 0 : expirationTimeSinceNow)
```

Comment:
```suggestion
            let expirationTimeout = UInt64(clamping: expirationTimeSinceNow)
```

Reply:
TIL :)

Reply:
Another option if we're only concerned about underflows.
```swift
let expirationTimeout = UInt64(max(0, expirationTimeSinceNow))
```